lona/html/abstract_node.py

```diff
@@ -102,6 +102,18 @@ def lock(self):
 
         return document.lock
 
+    # server data #############################################################
+    @property
+    def server_data(self):
+        if not hasattr(self, '_server_data'):
+            self._server_data = {}
```

Comment:
What about concurrency? Don't we need to protect `server_data` with lock?

Reply:
Oh good point! This would be the only user accessible data structure that is not thread safe. 
We can reuse ``server.state`` for this. Should we call it ``node.state`` in this case?

Reply:
I used ``lona.server_state.ServerState`` instead of an simple dict to make it thread-safe. In order to make the naming consistent i renamed it to ``node.state``